sympy/tensor/indexed.py

```diff
@@ -359,7 +259,10 @@ def args(self):
             return self._args
 
     def _hashable_content(self):
-        return Expr._hashable_content(self) + (self._shape,)
+        if self._shape:
+            return super(IndexedBase, self)._hashable_content() + (self._shape,)
+        else:
+            return super(IndexedBase, self)._hashable_content()
```

Comment:
Ah, that's new behaviour in Python 3 (or maybe just 3.4).
I opened issue #9332 for that.